docs/keyed_services.md

```diff
@@ -55,6 +55,22 @@ and pass in the services you require instead of calling the factory methods
 internally. This reduces dependencies on the factories and makes unit
 testing/mocking simpler.
 
+#### Keyed Services and Unit Tests
+
+If the KeyedService needs to be created automatically with the profile, then it
+should not be created during unit tests otherwise it is not possible to
+use a `TestProfileIOS` without this service (and then the tests 
```

Comment:
This also applies to `Profile` on desktop, it just isn't well documented like ios